src/alembic/versions/20251016_1000_7e5b5cf430ff_destructive_migration_filesystem_and_.py

```
@@ -140,62 +115,9 @@ def upgrade() -> None:
-        except Exception as e:
-            # If this fails, the restore still worked, just log it
-            logger.warning(f"Could not update alembic_version: {e}")
-            logger.log(
-                "DATABASE",
-                "   Database is restored, but alembic version tracking may be inconsistent",
-            )
-    else:
-        logger.error("❌ Failed to restore database from snapshot!")
-        raise Exception("Downgrade failed: Could not restore database from snapshot")
+    logger.error("Downgrade is not supported for this destructive migration.")
+    raise Exception("Downgrade is not supported for this destructive migration.")
```

Ruff (0.14.0)

Create your own exception

(TRY002)